### PR TITLE
Update to keyserver URL and reducing image layers

### PR DESCRIPTION
**Pull request by alek5k**
Please see [this thread](https://github.com/osrf/docker_images/pull/62) for a discussion on this pull request.

I have also changed the language environment variable declarations to be inline to prevent an additional image layer from being created.

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
The ROS2 docker image and templates are deprecated and don't accept any changes anymore. Is it possible to reduce this PR to target only the `create_ros_core_image.Dockerfile.em` file?

Unless I'm mistaken the snippet [add_distribution_repositories](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/ros_buildfarm/templates/snippet/add_distribution_repositories.Dockerfile.em) included [here](https://github.com/osrf/docker_templates/blob/master/ros_docker_images/templates/docker_images/create_base_image.Dockerfile.em#L13) should already add the ROS keys. any idea why the logic is duplicated in this file? @tfoote @ruffsl may have more info?

File: ros_docker_images/templates/docker_images/create_ros2_image.Dockerfile.em [hunk 1]
```diff
@@ -42,8 +42,7 @@ RUN apt-get update && apt-get install -q -y \
 RUN apt-get update && apt-get install -y \
     locales \
     && locale-gen en_US.UTF-8
-ENV LANG en_US.UTF-8
-ENV LC_ALL en_US.UTF-8
+ENV LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8
```
Comment (tfoote) [inline ros_docker_images/templates/docker_images/create_ros2_image.Dockerfile.em]: This is unrelated. If you'd like to propose changing this please do so in a separate PR with justification. I'll note that the best practices section of the user guide still uses the single line format: https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#env

**Comment by ruffsl** [comment]
I've made a new PR that just touches the kerserver URLs: see https://github.com/osrf/docker_templates/pull/19
@alek5k please check this works for your firewall

@mikaelarguedas , the official docker library practices define the style of using the keyserver, see the readme 
https://github.com/docker-library/official-images

I can make a future PR to template this via .em.

**Comment by mikaelarguedas** [comment]
> @mikaelarguedas , the official docker library practices define the style of using the keyserver, see the readme

I think that the argument about deduplicating logic is still valid. But it can be done in a later PR

**Comment by tfoote** [comment]
> The ROS2 docker image and templates are deprecated and don't accept any changes anymore. Is it possible to reduce this PR to target only the create_ros_core_image.Dockerfile.em file?

Although they are deprecated. I'd suggest keeping the url's up to date still makes sense if we're doing a general search and replace.

**Comment by mikaelarguedas** [comment]
> > The ROS2 docker image and templates are deprecated and don't accept any changes anymore. Is it possible to reduce this PR to target only the create_ros_core_image.Dockerfile.em file?

> Although they are deprecated. I'd suggest keeping the url's up to date still makes sense if we're doing a general search and replace.

Agreed in the case of modifying all files, that is what @ruffsl did in #19 :+1:

**Comment by alek5k** [comment]
@ruffsl I verified the hkp://ha url behind the firewall as per this pull request, but not the hkp://p80 address. Did you want me to verify p80 specifically?

File: ros_docker_images/templates/docker_images/create_ros2_image.Dockerfile.em (hunk 1 above)
Comment (alek5k) [inline ros_docker_images/templates/docker_images/create_ros2_image.Dockerfile.em]: @tfoote It just reduces the number of layers for the docker image. There is a limit to the number of layers (127 in total) and any small optimizations like this give the end user more flexibility. If you think this is unnecessary let me know and I wont bother with the PR.

Cheers

**Pull request closed without merging**